### PR TITLE
Automatically install bf for schema merge.

### PR DESCRIPTION
**Pull request by chrimc62**
test.schema has been having unnecessary churn and this fixes the issue so that it should only change when people change component schems.  This was for two reasons:
1) A bug in dialog:merge where oneOf clauses were not being sorted. (Which has been fixed and checked in.)
2) Developers running an older version of bf.  This is fixed by auto-installing the latest version if there is a change in test.schema.  

This PR also contains a refactoring of the tests so BF is only executed when the test is run and not when we open Test Explorer (addressses part of #4802)

### CODE REVIEW

File: tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/SchemaMergeTests.cs [hunk 1]
```diff
@@ -57,17 +55,35 @@ static SchemaMergeTests()
...
 
                     var process = Process.Start(startInfo);
                     var error = process.StandardError.ReadToEnd();
                     process.WaitForExit();
 
+                    var newSchema = File.Exists(schemaPath) ? File.ReadAllText(schemaPath) : string.Empty;
+                    if (error.Length != 0 || !newSchema.Equals(oldSchema))
+                    {
+                        // Try installing latest bf if the schema changed
+                        startInfo.Arguments = $"/C npm i -g @microsoft/botframework-cli@next";
```
Comment (EricDahlvang) [inline tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/SchemaMergeTests.cs:76]: This assumes npm is installed.

Although I understand the need for this, it feels odd to have an install within a test vs a build requirement.  (That said, I use the output from this test while working with composer... this is useful and convenient.)

File: tests/tests.schema [hunk 2]
```diff
@@ -3925,61 +3922,7 @@
           "$ref": "#/definitions/Microsoft.RegexRecognizer"
         },
         {
-          "$ref": "#/definitions/Microsoft.AgeEntityRecognizer"
```
Comment (EricDahlvang) [inline tests/tests.schema:3928]: I'm curious why these entity recognizers shouldn't be in the schema?

**Review by EricDahlvang — APPROVED** [review]

**Review by gabog — CHANGES_REQUESTED** [review]
This will help, but can we move this out of the static constructor? Just opening test explorer will cause this code to run (while VS tries to discover tests).

XUnit doesn't provide an equivalent of ClassInitialize and we need to create a fixture, some more info here: https://blog.somewhatabstract.com/2016/12/05/classinitialize-classcleanup-and-sharing-data-across-tests-in-xunit2/

File: tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/SchemaMergeTests.cs (hunk 1 above)
Comment (chrimc62) [reply tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/SchemaMergeTests.cs:76]: In order to get bf you already need to have npm installed and this is the only way I knew to ensure you have the latest.

File: tests/tests.schema (hunk 2 above)
Comment (chrimc62) [reply tests/tests.schema:3928]: They were not removed---just reordered into sorted order.

Comment (EricDahlvang) [reply tests/tests.schema:3928]: @chrimc62 they have not been reordered in this PR, they have been removed.  This is the definition of: Microsoft.IRecognizer:
https://github.com/microsoft/botbuilder-dotnet/blob/d1c7e93e173d71d15f119c4a5b08995fc1586078/tests/tests.schema#L3898

```json
"Microsoft.IRecognizer": {
      "title": "Microsoft recognizer",
      "description": "Components which derive from Recognizer class",
      "$role": "interface",
      "oneOf": [
        {
          "$ref": "#/definitions/Microsoft.CrossTrainedRecognizerSet"
        },
        {
          "$ref": "#/definitions/Microsoft.LuisRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.MultiLanguageRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.OrchestratorRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.QnAMakerRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.RecognizerSet"
        },
        {
          "$ref": "#/definitions/Microsoft.RegexRecognizer"
        },
        {
          "type": "string"
        }
      ]
    }
```

Should actually be:
```json
   "Microsoft.IRecognizer": {
      "title": "Microsoft recognizer",
      "description": "Components which derive from Recognizer class",
      "$role": "interface",
      "oneOf": [
        {
          "type": "string"
        },
        {
          "$ref": "#/definitions/Microsoft.OrchestratorRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.LuisRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.QnAMakerRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.CrossTrainedRecognizerSet"
        },
        {
          "$ref": "#/definitions/Microsoft.MultiLanguageRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.RecognizerSet"
        },
        {
          "$ref": "#/definitions/Microsoft.RegexRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.AgeEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.ChannelMentionEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.ConfirmationEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.CurrencyEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.DateTimeEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.DimensionEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.EmailEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.GuidEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.HashtagEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.IpEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.MentionEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.NumberEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.NumberRangeEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.OrdinalEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.PercentageEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.PhoneNumberEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.RegexEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.TemperatureEntityRecognizer"
        },
        {
          "$ref": "#/definitions/Microsoft.UrlEntityRecognizer"
        }
      ]
    }
```

**Review by gabog — APPROVED** [review]

**Review by mrivera-ms — APPROVED** [review]

**Pull request merged**